qemu/tests/boot_with_machine_types.py

```diff
@@ -0,0 +1,46 @@
+import logging
+
+from virttest import error_context
+from virttest import utils_misc
+from virttest import env_process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    boot with different machine type:
+    1) get supported machine type
+    2) boot guest with different machine types
+
+    :param test: kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    error_context.context("Get host's supported machine type", logging.info)
+    qemu_binary = utils_misc.get_qemu_binary(params)
+    tmp = utils_misc.get_support_machine_type(qemu_binary)
+    (supported_machine_types, _) = tmp
+    machine_types = []
+    for m_type in supported_machine_types:
+        if params["machine_type"] in m_type:
```

Comment:
Have filtered it via:
  for m_type in supported_machine_types:
        if params["machine_type"] == "pc":
            if "q35" not in m_type:
                machine_types.append(m_type)
        elif params["machine_type"] in m_type:
            machine_types.append(m_type)